xarray/core/weighted.py

```diff
@@ -239,6 +315,123 @@ def _weighted_std(
 
         return cast("DataArray", np.sqrt(self._weighted_var(da, dim, skipna)))
 
+    def _weighted_quantile(
+        self,
+        da: "DataArray",
+        q: Union[float, Iterable[float]],
+        dim: Union[Hashable, Iterable[Hashable]] = None,
+        skipna: Optional[bool] = None,
+        keep_attrs: Optional[bool] = None,
+    ) -> "DataArray":
+        """Apply a weighted ``quantile`` to a DataArray along some dimension(s)."""
+
+        def _get_h(n: float, q: np.ndarray, method: "QUANTILE_METHODS") -> np.ndarray:
```

Comment:
```suggestion
        def _get_h(n: float, q: np.ndarray, method: QUANTILE_METHODS) -> np.ndarray:
```

Reply:
pyupgrade should automagically fix this once  #6257 has been merged.